src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient_private.cs

```diff
@@ -127,56 +105,54 @@ Uri BuildUriForLocksRoute(string key, string label)
             return builder.Uri;
         }
 
-        void BuildBatchQuery(UriBuilder builder, BatchRequestOptions options)
+        void BuildBatchQuery(UriBuilder builder, SettingSelector selector)
         {
-            if (!string.IsNullOrEmpty(options.Key))
-            {
-                builder.AppendQuery(KeyQueryFilter, options.Key);
-            }
-
-            if (!string.IsNullOrEmpty(options.BatchLink))
+            if (selector.Keys.Count > 0)
             {
-                builder.AppendQuery("after", options.BatchLink);
+                var keys = string.Join(",", selector.Keys).ToLower();
+                builder.AppendQuery(KeyQueryFilter, keys);
             }
 
-            if (options.Label != null)
+            if (selector.Labels.Count > 0)
             {
-                if (options.Label == string.Empty)
+                for(int i =0; i < selector.Labels.Count; i++)
                 {
-                    options.Label = "\0";
+                    if (selector.Labels[i] == string.Empty)
+                    {
+                        selector.Labels[i] = "\0";
```

Comment:
Why is zero character required?

Reply:
Part of the service specification: " To explicitly reference a key-value without a label use "\0" "
https://msazure.visualstudio.com/Azure%20AppConfig/_git/AppConfigService?path=%2Fdocs%2Fprotocol%2Fkv.md&version=GBmaster

Reply:
Oh wow, ok.

I think you should avoid mutating customers objects at this method though.